sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -0,0 +1,730 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.implementation.handler;
+
+import com.azure.core.amqp.AmqpRetryOptions;
+import com.azure.core.amqp.AmqpRetryPolicy;
+import com.azure.core.amqp.exception.AmqpErrorCondition;
+import com.azure.core.amqp.exception.AmqpErrorContext;
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.amqp.implementation.ExceptionUtil;
+import com.azure.core.amqp.implementation.ReactorDispatcher;
+import com.azure.core.amqp.implementation.RetryUtil;
+import com.azure.core.util.logging.ClientLogger;
+import org.apache.qpid.proton.amqp.messaging.Outcome;
+import org.apache.qpid.proton.amqp.messaging.Rejected;
+import org.apache.qpid.proton.amqp.messaging.Released;
+import org.apache.qpid.proton.amqp.transaction.TransactionalState;
+import org.apache.qpid.proton.amqp.transport.DeliveryState;
+import org.apache.qpid.proton.amqp.transport.DeliveryState.DeliveryStateType;
+import org.apache.qpid.proton.amqp.transport.ErrorCondition;
+import org.apache.qpid.proton.engine.Delivery;
+import reactor.core.Disposable;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.MonoSink;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.StringJoiner;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.azure.core.amqp.implementation.ClientConstants.DELIVERY_STATE_KEY;
+import static com.azure.core.util.FluxUtil.monoError;
+
+/**
+ * Manages the received deliveries which are not settled on the broker. The client can later request settlement
+ * of each delivery by sending a disposition frame with a state representing the desired-outcome,
+ * which the application wishes to occur at the broker. The broker acknowledges this with a disposition frame
+ * with a state (a.k.a. remote-state) representing the actual outcome (a.k.a. remote-outcome) of any work
+ * the broker performed upon processing the settlement request and a flag (a.k.a. remotely-settled) indicating
+ * whether the broker settled the delivery.
+ */
+public final class ReceiverUnsettledDeliveries implements AutoCloseable {
```

Comment:
This class and any public methods are to be package internal in phase-2; once ReceiveLinkHandler abstract the disposition